### ISSUE TITLE
Implement Pagination for API Keys Table

### ISSUE THREAD

**Issue by unrenamed** [unkeyed/unkey#2540]
### Preliminary Checks

- [X] I have reviewed https://unkey.com/docs for existing features that would solve my problem

- [X] I have searched for existing feature requests: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a feature request directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Is your feature request related to a problem? Please describe.

Yes, the feature request is related to a significant usability problem. Currently, the API Keys table displays only the first page of results, limiting users' ability to access or view the remaining 400 keys from a sample of 500. This can lead to frustration and inefficiencies when users need to manage, review, or analyze a larger set of keys, as they cannot navigate beyond the initial page.

### Describe the solution

The proposed solution is to implement a pagination feature for the API Keys table. This would involve one of the following approaches:

1. Pagination Controls
    Adding controls (such as "Previous," "Next," and page numbers) to navigate through the different pages of keys.
2. Dynamic Loading
     Allowing the table to load a specific number of keys per page (e.g., 10, 20, or 50), providing a manageable view of the data.  _Given the existing `cursor`-based pagination in the API, this may be the most optimal approach_.

### Describe alternatives you have considered (if any)

Alternatives considered include:

1. Infinite Scrolling: Automatically load more keys as users scroll. This could complicate navigation.
2. Search and Filter Options: Enhance search functionalities to find specific keys quickly, but this doesn't resolve the viewing issue.

### Additional context

Uploading Screen Recording 2024-10-22 at 21.54.31.mov…

**Comment by chronark** [comment]
This entire page needs to be redesigned, we're currently planning this.

**Comment by BadriVishalPadhy** [comment]
@chronark  should i add pagination to this?

**Comment by GARY121github** [comment]
@chronark can I work on this issue if no one is working on it?

**Comment by Surajsuthar** [comment]
@chronark  can i go with this??

**Comment by GARY121github** [comment]
@rohitlohar45 I don't think you can work on this issue because the core team is working on this issue.

**Comment by perkinsjr** [comment]
This was completed in the rewrite
